doc/develop/test/twister.rst

```diff
@@ -823,6 +823,34 @@ line break instead of white spaces.
 
 Most everyday users will run with no arguments.
 
+Selecting platform scope:
+*************************
+
+One of the key features of Twister is its ability to decide on which platforms a given
+test scenario should run. This behavior has its roots in Twister being developed as
+a test runner for Zephyr's CI. With hundreds of available platforms and thousands of
+tests, the testing tools should be able to adapt the scope and select/filter out what
+is relevant and what is not.
+
+Twister is always preparing an initial list of platforms in scope for a given test,
+based on command line arguments and the test's configuration file. Then, platforms
+that don't fulfill the required conditions are filtered out from the scope.
+Using ``--force-platform`` turns off filtering.
```

Comment:
This is not clear to me. What are "the required conditions" exactly? What does turning the filtering off mean?

If a platform is in my `platform_allow` list but not in the default platform list, will it be run if I use `--force-platform`?

Also, the twister CLI help say the following:

```
Force testing on selected platforms, even if they are excluded in the test configuration (testcase.yaml).
```

It doesn't sound the same as what you describe to me. From what I understand it's just a way to ignore `platform_exclude`.

Reply:
`--force-platform` allows you to run test on a platform that is not listed in a `platform_allow` list. This flag is rather used to tests / debug platforms, but finally tested platform should be added to `platform_allow` list, if you want to have it in regression.

Reply:
I added more clarification about "required conditions" and "filtering" and "--force-platform".

> If a platform is in my platform_allow list but not in the default platform list, will it be run if I use --force-platform?
It depends on what other arguments you've used.  

I believe all logic required to answer this is presented in the PR. I tried to explain multiple times (it is also already part of the docs at https://docs.zephyrproject.org/latest/develop/test/twister.html#tests at `platform_allow`) that platform_allow shouldn't be treated as a scope defining parameter but a filter. Therefore, I am against describing this specific combinations since IMO it is legitimizing misuse of platform_allow.